### PR TITLE
Build eksctl docker image on release (don't yet push)

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: Makefile.docker [hunk 1]
```diff
@@ -12,9 +12,11 @@ image_manifest_file = build/docker/build_image_manifest
 build_image_tag_file = build/docker/image_tag
 build_image_tag = $(shell cat $(build_image_tag_file))
 
-build_image_name = weaveworks/eksctl-build:$(build_image_tag)
+build_image = weaveworks/eksctl-build
+build_image_name = $(build_image):$(build_image_tag)
 
-intermediate_container_name = eksctl-build-$(unique_tag)
+eksctl_image = weaveworks/eksctl
+eksctl_image_name = "$(eksctl_image):$${EKSCTL_IMAGE_VERSION}"
```
Comment (martina-if) [inline Makefile.docker:19]: >$${EKSCTL_IMAGE_VERSION}

Double `$`?

Comment (michaelbeaumont) [reply Makefile.docker:19]: Makefile magic `$$` = `$` so it ends up an env variable in the shell command

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+ARG BUILD_IMAGE=/weaveworks/eksctl-build:0849e149eca6668d7e1a5b400dbbbb5cac650e52
+FROM $BUILD_IMAGE as build
```
Comment (martina-if) [inline Dockerfile:2]: Do we need the build image as base or can we just use a distroless?

Comment (michaelbeaumont) [reply Dockerfile:2]: the final image is distroless:
https://github.com/weaveworks/eksctl/pull/2511/files#diff-3254677a7917c6c01f55212f86c57fbfR19

File: Makefile.docker (hunk 1 above)
Comment (martina-if) [reply Makefile.docker:19]: ahaaa, thanks!

**Pull request merged**